### PR TITLE
DetectX Module

### PR DESCRIPTION
**Pull request by WardsParadox**
Will pull a json file from DetectX granted the admin deploying this module places it in the cache dir.

### CODE REVIEW

File: app/modules/detectx/scripts/install.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+# detectx controller
+#CTL="${BASEURL}index.php?/module/detectx/"
+	# Set preference to include this file in the preflight check
```
Comment (bochoven) [inline app/modules/detectx/scripts/install.sh]: Please also drop an empty json so the client does not complain about the file not found

**Comment by bochoven** [comment]
Thanks. Could you please fix the indenting? I have a hard time reading the code.. ;-D

**Comment by WardsParadox** [comment]
Atom must be messing with it since I turned off softwraps. I'll get it all prettified and a touch statement to add an empty json file in `install.sh` then commit again.

**Comment by WardsParadox** [comment]
@bochoven Is that any better?

File: app/modules/detectx/detectx_model.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		// Check if data was uploaded
+		if ( ! $json ){
+			throw new Exception("Error Processing Request: No JSON file found", 1);
+		}
+
+		// Delete previous set
+		$this->deleteWhere('serial_number=?', $this->serial_number);
+
+		// Process json into object thingy
+    $data = json_decode($json, true);
```
Comment (bochoven) [inline app/modules/detectx/detectx_model.php]: This block needs indenting

File: app/modules/detectx/locales/en.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  "listing":{
+    "reporttitle": "DetectX Report",
+    "clienttitle": "DetectX",
+    "clienttab": "DetectX",
+    "issues": "Issues",
+    "numberofissues": "Number of Issues",
+    "searchdate": "Search Date",
+    "status": "Status"
+},
+"widget": {
```
Comment (bochoven) [inline app/modules/detectx/locales/en.json]: Please also indent the JSON

**Comment by WardsParadox** [comment]
@bochoven should be fixed now.

File: app/modules/detectx/README.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+<plist version="1.0">
+<dict>
+	<key>Label</key>
+	<string>DetectX All User Run</string>
+	<key>ProgramArguments</key>
+	<array>
+		<string>/Applications/DetectX Swift.app/Contents/MacOS/DetectX Swift</string>
+		<string>search</string>
+		<string>-aj</string>
+		<string>/usr/loca/munki/preflight.d/cache/detectx.json</string>
```
Comment (bochoven) [inline app/modules/detectx/README.md]: Path is missing an 'l', should be `/usr/local/munki/preflight.d/cache/detectx.json`

File: app/modules/detectx/provides.php [hunk 5]
```diff
@@ -0,0 +1,14 @@
+<?php
+//
+return array(
+    'client_tabs' => array(
+        'detectx-tab' => array('view' => 'detectx_tab', 'i18n' => 'detectx.clienttitle', 'badge' => 'detectx-cnt'),
```
Comment (bochoven) [inline app/modules/detectx/provides.php]: detectx.clienttitle does not exist in the locale.json. You may want to change that to detectx.title (this controls the name in the dropdown in the client detail page)

File: app/modules/detectx/provides.php [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+    ),
+    'listings' => array(
+       'detectx' => array('view' => 'detectx_listing', 'i18n' => 'detectx.clienttitle'),
```
Comment (bochoven) [inline app/modules/detectx/provides.php]: detectx.clienttitle does not exist in the locale.json. You may want to change that to detectx.title (this controls the name in the dropdown in the listing menu)

File: app/modules/detectx/detectx_model.php [hunk 7]
```diff
@@ -0,0 +1,67 @@
+<?php
+class Detectx_model extends \Model {
+
+	function __construct($serial='')
+	{
+		parent::__construct('id', 'detectx'); //primary key, tablename
+		$this->rs['id'] = '';
+		$this->rs['serial_number'] = $serial;
```
Comment (bochoven) [inline app/modules/detectx/detectx_model.php]: Please add a UNIQUE rt statement: `$this->rt['serial_number'] = 'VARCHAR(255) UNIQUE';`

File: app/modules/detectx/detectx_model.php [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    		$this->rs['status'] = '';
+		$this->rs['issues'] = ''; $this->rt['issues'] = 'TEXT';
+
+
+		// Schema version, increment when creating a db migration
+		$this->schema_version = 0;
+
+		// Add indexes
+    		$this->idx[] = array('numberofissues');
+    		$this->idx[] = array('status');
```
Comment (bochoven) [inline app/modules/detectx/detectx_model.php]: I would add an index on search date too.

File: app/modules/detectx/views/detectx_listing.php [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+            <th data-i18n="serial" data-colname='reportdata.serial_number'></th>
+            <th data-i18n="detectx.listing.searchdate" data-colname='detectx.searchdate'></th>
+            <th data-i18n="detectx.listing.status" data-colname='detectx.status'></th>
+            <th data-i18n="detectx.listing.numberofissues" data-colname='detectx.numberofissues'></th>
+          </tr>
+        </thead>
+
+        <tbody>
+          <tr>
+            <td data-i18n="listing.loading" colspan="10" class="dataTables_empty"></td>
```
Comment (tuxudo) [inline app/modules/detectx/views/detectx_listing.php]: colspan= should be equal to the total amount of columns in the listing.

**Comment by WardsParadox** [comment]
@bochoven @tuxudo Thanks for pointing those out, should be fixed now.

**Pull request merged**